node/store.go

```diff
@@ -74,5 +74,12 @@ func NewStore(opts ...StoreOpt) (datastore.RootStore, error) {
 	badgerOpts.ValueLogFileSize = options.valueLogFileSize
 	badgerOpts.EncryptionKey = options.encryptionKey
 
+	if len(options.encryptionKey) > 0 {
```

Comment:
suggestion: I prefer `if options.encryptionKey != ""` as it clearly shows that the type of the field is string, not a slice

Reply:
I think you may have misread the type. `encryptionKey` is a `[]byte`